src/super_gradients/training/utils/visualization/depth_estimation.py

```diff
@@ -0,0 +1,39 @@
+import cv2
+import numpy as np
+
+
+class DepthVisualization:
+    @staticmethod
+    def process_depth_map_for_visualization(
+        depth_map: np.ndarray, color_scheme: int = cv2.COLORMAP_VIRIDIS, drop_extreme_percentage: float = 0, inverse: bool = False
+    ) -> np.ndarray:
+        """
+        Process a depth map for visualization.
+
+        This method enhances the visual representation of a depth map by:
+        1. Clipping extreme values based on the specified percentage.
+        2. Normalizing the depth map to the 0-255 range.
+        3. Optionally inverting the depth map.
+        4. Applying a color map using OpenCV's applyColorMap.
+
+        :param depth_map:               Input depth map as a NumPy array.
+        :param color_scheme:            OpenCV color scheme for the depth map visualization.
+        :param drop_extreme_percentage: Percentage of extreme values to drop.
+        :param inverse:                 Apply inversion (255 - depth_map) if True.
+
+        :return:                        Processed colormap of the depth map for visualization.
+        """
+
+        # Drop extreme values
+        min_val = np.percentile(depth_map, drop_extreme_percentage)
+        max_val = np.percentile(depth_map, 100 - drop_extreme_percentage)
+        depth_map = np.clip(depth_map, min_val, max_val)
+
+        # Normalize to 0-255
+        depth_map_normalized = ((depth_map - min_val) / (max_val - min_val) * 255).astype(np.uint8)
+        if inverse:
```

Comment:
A few important aspects:
1) I beleive inversion should be done as 1 /  depth, this is how disparity map is defined
2) Usually when using direct depth for visualization a COLORMAP_VIRIDIS is fine choice
3) For disparity (inverse=True) a COLORMAP_MAGMA is a default choice (IDK why)
4) OpenCV uses BGR notation, so I think we should to BGR->RGB conversion at the very end

Reply:
3d38e6a39442e075789f4b1fdb95fdffe2f135d7 see updated